packages/lib/src/vis/surface/SurfaceMesh.tsx

```diff
@@ -0,0 +1,58 @@
+import { getDims } from '@h5web/shared';
+import { useThree } from '@react-three/fiber';
+import { useLayoutEffect, useState } from 'react';
+import { BufferGeometry } from 'three';
+
+import { useDataToColorScale } from '../hooks';
+import GlyphMaterial from '../line/GlyphMaterial';
+import { GlyphType } from '../line/models';
+import SurfaceShaderMaterial from './SurfaceShaderMaterial';
+import type { SurfaceVisProps } from './SurfaceVis';
+import { useBufferAttributes } from './hooks';
+
+type Props = Required<SurfaceVisProps>;
+
+function SurfaceMesh(props: Props) {
+  const { dataArray, domain, colorMap, invertColorMap, scaleType, showPoints } =
+    props;
+
+  const { rows, cols } = getDims(dataArray);
+
+  const [dataGeometry] = useState(() => new BufferGeometry());
+  const invalidate = useThree((state) => state.invalidate);
+
+  const dataToColorScale = useDataToColorScale(
+    scaleType,
+    domain,
+    colorMap,
+    invertColorMap
+  );
+
+  const {
+    position,
+    color,
+    index: indices,
+  } = useBufferAttributes(dataArray, dataToColorScale);
```

Comment:
For now, I make use of `BufferAttributes` to set the `position` and the `color`, following the `ScatterPoints` component.

I may change this later to use shaders instead for the color, similarly to `HeatmapMesh`